### PR TITLE
Obc/mobdensityupdate

### PR DESCRIPTION
**Pull request by OldBeardedCoder**
Fixed density overlap in old code, which was squaring density for nightmare, no increase for hell.  Upped density max to 45 and split between MonStats.txt & Levels.txt so that the density caps are not hit.

### CODE REVIEW

File: cfg.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
     "OutputDir": "",
     "MeleeSplash": true,
     "IncreaseStackSizes": true,
-    "IncreaseMonsterDensity": 1,
+    "IncreaseMonsterDensity": 8,
```
Comment (tlentz) [inline cfg.json:7]: leave this at 1 as the default

File: internal/d2mod/monsterdensity/monsterdensity.go [hunk 2]
```diff
@@ -5,41 +5,103 @@ import (
...
+		// Increased slider max to 45, then split the the value back down into 2 separate multipliers, the Grp multiplier and the MonDen multiplier, of which
+		// the Grp multiplier caps out at 4.5 and the MonDen multiplier caps out at 10.
+		// SuperUniques highest vanilla value/cap is 15/255.  Since (255/15 = 17) 17 is < 45, past 17x the density of superunique packs for the
+		// most dense area will stop increasing.  The cap of 99 may be purely artificial, however in 1.06 & 1.07 increasing the density past 99 would cause
+		// d2 to crash.
+		grpMult = monUMult / 4.5
+		maxDensityMult = monUMult / grpMult // 10 * 4.5 is about 45
+	}
+
+	//maxDensity := 10000
```
Comment (tlentz) [inline internal/d2mod/monsterdensity/monsterdensity.go:39]: unused?

File: internal/d2mod/monsterdensity/monsterdensity.go [hunk 3]
```diff
@@ -5,41 +5,103 @@ import (
...
+			if f.Rows[rowIdx][monstats.Primeevil] == "1" {
+				continue
+			}
+			if f.Rows[rowIdx][monstats.Npc] == "1" {
+				continue
+			}
+			if f.Rows[rowIdx][monstats.Align] != "" {
+				continue
+			}
+			{
```
Comment (tlentz) [inline internal/d2mod/monsterdensity/monsterdensity.go:92]: im unfamiliar with this syntax, what do these braces do?

**Review by tlentz — COMMENTED** [review]
comments

File: internal/d2mod/monsterdensity/monsterdensity.go (hunk 3 above)
Comment (OldBeardedCoder) [reply internal/d2mod/monsterdensity/monsterdensity.go:92]: I shouldn't have done that.  The extra braces enable 2 different colIdx variables due to forcing a different scope.   The fix is remove the braces and fix the second colIdx  := to be coIdx =
Do I remove it or do you?

File: cfg.json (hunk 1 above)
Comment (OldBeardedCoder) [reply cfg.json:7]: Agreed.

File: internal/d2mod/monsterdensity/monsterdensity.go (hunk 2 above)
Comment (OldBeardedCoder) [reply internal/d2mod/monsterdensity/monsterdensity.go:39]: It's the first one that's unused, the maxDensityMult = 4.5 that needs to be removed.  I was refactoring/cleaning up/moving stuff around and missed that.

**Review by tlentz — APPROVED** [review]

**Comment by tlentz** [comment]
LGTM

**Review by tlentz — APPROVED** [review]

**Pull request merged**